### PR TITLE
test: adding support to specify the operator and cluster namepaces

### PR DESCRIPTION
**Pull request by rkachach**
Adding support to specify operator and cluster namespaces as new command-line arguments for the `create-dev-cluster` script. This PR adds new arguments:

-  `-c`  to specify the cluster namespace
-  `-o` to specify the operator namespace
 
In addition, the script will automatically update the namespace in the corresponding YAML files. Operator namespace is create automatically if it doesn't exist.

**Testing:** 

**1- Create a cluster**
```
./tests/scripts/create-dev-cluster.sh -m -r -f -p minikube -c rook-cluster-ns -o rook-operator-ns
Using '/home/<user>/workspaces/src/github.com/rook/rook/tests/scripts/../../deploy/examples/' as examples directory..
Using rook-cluster-ns as cluster namespace..
Using rook-operator-ns as operator namespace..
Updating namespace on common.yaml
Updating namespace on operator.yaml
....


```

**2- Check namespaces are created correctly**
```
> kubectl get ns
NAME               STATUS   AGE
default            Active   21m
kube-node-lease    Active   21m
kube-public        Active   21m
kube-system        Active   21m
rook-cluster-ns    Active   21m
rook-operator-ns   Active   21m
```

**3- check pods from the operator and cluster namespaces are in Running state**

```
> kubectl get pods -n rook-operator-ns                                                    
NAME                                            READY   STATUS    RESTARTS   AGE
csi-cephfsplugin-provisioner-7568bf858f-m2ljp   5/5     Running   0          21m
csi-cephfsplugin-wjkhs                          2/2     Running   0          21m
csi-rbdplugin-provisioner-8db6c664-b2926        5/5     Running   0          21m
csi-rbdplugin-zth5c                             2/2     Running   0          21m
rook-ceph-operator-b89ccd545-bbcns              1/1     Running   0          22m
rook-discover-bgkgd                             1/1     Running   0          22m
```

```
> kubectl get pods -n rook-cluster-ns                                                               
NAME                                                 READY   STATUS      RESTARTS      AGE
prometheus-rook-prometheus-0                         3/3     Running     1 (19m ago)   19m
rook-ceph-crashcollector-minikube-7fd544c5d8-m7c7r   1/1     Running     0             13m
rook-ceph-exporter-minikube-65bc6b744b-fb9s4         1/1     Running     0             13m
rook-ceph-mds-myfs-a-88cc4df9c-r6k4g                 1/1     Running     0             13m
rook-ceph-mds-myfs-b-76785874b-kx26j                 1/1     Running     0             13m
rook-ceph-mgr-a-5bbc45bb75-ftx29                     1/1     Running     0             9m17s
rook-ceph-mon-a-5cf964b6f7-x87vs                     1/1     Running     0             21m
rook-ceph-nfs-my-nfs-a-c65d4c5d6-qgfzp               2/2     Running     0             13m
rook-ceph-osd-0-76478b97cd-ss2f8                     1/1     Running     0             20m
rook-ceph-osd-1-6db747f9b-bbjvt                      1/1     Running     0             20m
rook-ceph-osd-2-54f8d798b6-tjgjs                     1/1     Running     0             20m
rook-ceph-osd-prepare-minikube-ps6hv                 0/1     Completed   0             8m57s
rook-ceph-rgw-my-store-a-6797c9c47d-56jq4            1/1     Running     0             13m
rook-ceph-tools-757999d6c7-zpvfd                     1/1     Running     0             23m

```

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: tests/scripts/create-dev-cluster.sh [hunk 1]
```diff
@@ -101,41 +128,45 @@ check_examples_dir() {
...
 
 show_usage() {
     echo ""
     echo " Usage: $(basename "$0") [-r] [-m] [-p <profile-name>] [-d /path/to/rook-examples/dir]"
+    echo "  -f                Force cluster creation by deleting minikube profile"
     echo "  -r                Enable rook orchestrator"
     echo "  -m                Enable monitoring"
     echo "  -p <profile-name> Specify the minikube profile name"
     echo "  -d value          Path to Rook examples directory (i.e github.com/rook/rook/deploy/examples)"
+    echo "  -n <cluster-namespace>"
```
Comment (obnoxxx) [inline tests/scripts/create-dev-cluster.sh:168]:
```suggestion
    echo "  -c <cluster-namespace>"
```

**Review by obnoxxx — CHANGES_REQUESTED** [review]
Good change overall, but I found a typo in the usage text.

File: tests/scripts/create-dev-cluster.sh [hunk 2]
```diff
@@ -101,41 +128,45 @@ check_examples_dir() {
...
     done
 }
 
 enable_rook_orchestrator() {
     echo "Enabling rook orchestrator"
-    $KUBECTL rollout status deployment rook-ceph-tools -n rook-ceph --timeout=30s
-    $KUBECTL -n rook-ceph exec -it deploy/rook-ceph-tools -- ceph mgr module enable rook
-    $KUBECTL -n rook-ceph exec -it deploy/rook-ceph-tools -- ceph orch set backend rook
-    $KUBECTL -n rook-ceph exec -it deploy/rook-ceph-tools -- ceph orch status
+    $KUBECTL rollout status deployment rook-ceph-tools -n "$ROOK_CLUSTER_NS" --timeout=90s
```
Comment (obnoxxx) [inline tests/scripts/create-dev-cluster.sh:141]: @rkachach , This change from 30s to 90s timeout seems unrelated to the introduction of cmdline parameters for namespaces. Shouldn't it therefore be a separate commit and maybe even a separate PR?

**Review by obnoxxx — CHANGES_REQUESTED** [review]
Thanks for fixing the usage text, but one more issue. It is more a question/suggestion than a request to separate the change of timeout value from thge main commit.

File: tests/scripts/create-dev-cluster.sh (hunk 2 above)
Comment (rkachach) [reply tests/scripts/create-dev-cluster.sh:141]: @obnoxxx you are right, I'll put it as  separate commit because it's very small change and doesn't deserve its own PR :)

**Review by obnoxxx — APPROVED** [review]
Thanks for splitting the commit! LGTM!

**Comment by BlaineEXE** [comment]
Generally, with basic rook installs, the operator and cluster namespaces are the same. Is this something you are needing to do in development that this complexity is helpful?

**Comment by rkachach** [comment]
> Generally, with basic rook installs, the operator and cluster namespaces are the same. Is this something you are needing to do in development that this complexity is helpful?

@BlaineEXE the short answer is yes :) .. basically I needed that support when investigating and trying to fix some issues on the dashboard related of having difference namespace for cluster and/or operator such as https://github.com/rook/rook/issues/11883 and https://github.com/rook/rook/issues/11796. So having the possibility to specify the namespaces in those cases is really handy.
